.pre-commit-config.yaml

```diff
@@ -3,51 +3,30 @@ repos:
   rev: v4.6.0
   hooks:
   - id: check-merge-conflict
-  - id: end-of-file-fixer
-  - id: fix-encoding-pragma
-  - id: mixed-line-ending
-  - id: trailing-whitespace
   - id: check-added-large-files
     args: ['--maxkb=2000']
 
-# Sort package imports alphabetically
-- repo: https://github.com/PyCQA/isort
-  rev: 5.13.2
-  hooks:
-  - id: isort
-    args: ["--profile", "black", "--filter-files"]
-
-# Convert relative imports to absolute imports
-- repo: https://github.com/MarcoGorelli/absolufy-imports
-  rev: v0.3.1
-  hooks:
-  - id: absolufy-imports
+# Run ruff to lint and format
+- repo: https://github.com/astral-sh/ruff-pre-commit
+  # Ruff version.
+  rev: v0.4.4
+  hooks:
+    # Run the linter.
+  - id: ruff
+    args: [--fix]
+    # Run the formatter.
+  - id: ruff-format
 
 # Find common spelling mistakes in comments and docstrings
 - repo: https://github.com/codespell-project/codespell
   rev: v2.2.6
   hooks:
   - id: codespell
-    args: ['--ignore-regex="(\b[A-Z]+\b)"', '--ignore-words=.codespell.ignore'] # Ignore capital case words, e.g. country codes
+    # Ignore capital case words, e.g. country codes
+    args: ['--ignore-regex="(\b[A-Z]+\b)"', '--ignore-words=.codespell.ignore']
     types_or: [python, rst, markdown]
     files: ^(scripts|doc)/
 
-# Make docstrings PEP 257 compliant
-- repo: https://github.com/PyCQA/docformatter
-  rev: v1.7.5
-  hooks:
-  - id: docformatter
-    args: ['--in-place', '--make-summary-multi-line', '--pre-summary-newline']
-
-# Formatting with "black" coding style
-- repo: https://github.com/psf/black
-  rev: 24.4.2
-  hooks:
-  # Format Python files
-  - id: black
-  # Format Jupyter Python notebooks
-  - id: black-jupyter
-
```

Comment:
that means ruff replaces docformatter, balck and isort?

Reply:
Yes. For the Docstrings it covers mostly the same on what docformatter did and more (see [here](https://docs.astral.sh/ruff/rules/#pycodestyle-e-w)). I adjusted the rules in `pyproject.toml` and we can add some more in future (e.g. once we have docstrings everywhere, so it fails if you wanna merge something without docstring)

Reply:
great